transformers/DelegateContractTransformer.js

```diff
@@ -0,0 +1,77 @@
+const { getExtractorName } = require("spock-etl/lib/core/processors/extractors/instances/rawEventBasedOnTopicExtractor");
+const { handleEvents } = require("spock-etl/lib/core/processors/transformers/common");
+const { getTxByIdOrDie } = require("spock-etl/lib/core/processors/extractors/common");
+const BigNumber = require("bignumber.js").BigNumber;
+const ethers = require("ethers");
+
+// @ts-ignore
+const abi = require("../abis/vote_delegate_contract.json");
+
+const extractor_name = 'delegate';
+module.exports.DELEGATE_EXTRACTOR_NAME = extractor_name;
+
+module.exports.default = () => ({
+  name: `DelegateContractTransformer`,
+  dependencies: [getExtractorName(extractor_name)],
+  transform: async (services, logs) => {
+    const validData = logs[0].every(l => l.data && l.data !== '0x');
+    if (validData) await handleEvents(services, abi, logs[0], handlers);
+  },
+});
+
+const handlers = {
+  async Lock(services, info) {
+    const tx = await getTxByIdOrDie(services, info.log.tx_id);
```

Comment:
Maybe fetching the tx below the check for `.chief()` would save us a call to the db in the case of non-delegate contracts. Now sure how much impact it would have tbh, but it might streamline a little bit

Reply:
Good idea 👍